src/town_map.h

```diff
@@ -74,6 +74,28 @@ inline void SetHouseType(Tile t, HouseID house_id)
 	SB(t.m8(), 0, 12, house_id);
 }
 
+/**
+ * Check if the house is protected from removal by towns and AIs.
```

Comment:
I really do not understand the "by towns and AIs" part. Where in this code does it limit it to those two? Can other other companies in multiplayer games remove it? I really miss something here, but this "by towns and AIs" feels wrong. I would expect it is protected for everyone to remove it. And only magic bulldozer can undo that. Or possibly the original owner. But I cannot see code of that here. I might be missing something crucial :)